libtenzir/include/tenzir/plugin.hpp

```diff
@@ -1126,6 +1142,11 @@ extern const char* TENZIR_PLUGIN_VERSION;
     }                                                                          \
     extern "C" auto tenzir_libtenzir_build_tree_hash() -> const char* {        \
       return ::tenzir::version::build::tree_hash;                              \
+    }                                                                          \
+    extern "C" auto tenzir_plugin_dependencies() -> const char* const* {       \
+      static const char* dependencies[]                                        \
+        = {TENZIR_PLUGIN_DEPENDENCIES_NULLSEP};                                \
+      return dependencies;                                                     \
```

Comment:
An alternative, to avoid going through `std::vector`, but might be more trouble than it's worth. Take your pick.

```
static const auto dependencies = []<typename... T>(T... deps) {
  if constexpr (sizeof...(T) == 0) {
    return std::array<const char*, 1>{nullptr};
  } else {
    return std::array<const char*, sizeof...(T) + 1>{deps..., nullptr};
  }
}(DEPS);
return dependencies.data();
```